packages/common-lib/src/components/layout/AppBar.js

```diff
@@ -43,12 +59,59 @@ export default function AppBar({
     if (setSearchState) setSearchState(boolean)
     setSearchInput(boolean)
   }
+  const [pinnedAnnouncementsList, setPinnedAnnouncementsList] = React.useState(
+    pinnedAnnouncementsData
+  )
+  const pinnedData = React.useMemo(
+    () => pinnedAnnouncementsList,
+    [pinnedAnnouncementsList]
+  )
 
   return (
     <Box pt={7} px={5} {...props?._box}>
       <StatusBar bg='gray.600' barStyle='light-content' />
       <Box safeAreaTop bg='gray.600' />
+      {showPinnedAnnouncements ? (
```

Comment:
this can be extracted out to PinnedAnnoumcementsComponent and should be passed from Application similar to LoginComponent

Reply:
Extracted. Please check